components/script/dom/gpudevice.rs

```diff
@@ -320,6 +416,16 @@ impl GPUDeviceMethods for GPUDevice {
             }
         });
 
+        let mut scope_id = None;
+        if let Some(s_id) = self.scope_stack.borrow_mut().last() {
```

Comment:
this should probably be abstracted away and become `let scope_id = self.use_current_scope();`